plugins/nft/nft-transactions/src/transactions/handlers/nft-mint.ts

```diff
@@ -44,22 +44,14 @@ export class NftMintTransactionHandler extends Handlers.TransactionHandler {
         walletManager: State.IWalletManager,
     ): Promise<void> {
         const { tokenId, properties } = getCurrentNftAsset(transaction.data.asset);
-
+        const nftManager = app.resolvePlugin<NftsManager>("core-nft");
         // check if token is already owned
-        if (
-            !!walletManager.allByAddress().find(wallet => {
-                if (wallet.hasAttribute("tokens")) {
-                    return Object.keys(wallet.getAttribute<INftWalletAttributes>("tokens")).includes(tokenId);
-                }
-                return false;
-            })
-        ) {
+        if (await nftManager.exists(tokenId)) {
```

Comment:
What is the performance impact?
Especially when starting a node from scratch?

Reply:
benched on my local machine:
old version (parse all wallets in ram): 21.223ms
new version (db query) : 4.088ms